vendor/src/github.com/docker/libnetwork/endpoint_cnt.go

```diff
@@ -108,6 +108,25 @@ func (ec *endpointCnt) EndpointCnt() uint64 {
 	return ec.Count
 }
 
+func (ec *endpointCnt) updateStore() error {
+retry:
```

Comment:
is there no backoff or timeout whatsoever for this? what happens when the store is down and this code keeps failing and retrying? is this going to cause a stack overflow?